### ISSUE TITLE
onMarkerTap not called

### ISSUE THREAD

**Issue by starshipcoder** [lpongetti/flutter_map_marker_cluster#153]
Since version 1.0.2, MarkerClusterLayerOptions.onMarkerTap callback is no more called

**Comment by RicardoRB** [comment]
Same here

**Comment by Hamitix** [comment]
Same here :(

**Comment by Pepslee** [comment]
@starshipcoder @RicardoRB @Hamitix  can you share you flutter --version output ?
I have an assumption, that the problem not in cluster version but in flutter version.

**Comment by starshipcoder** [comment]
Flutter 3.7.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 9944297138 (13 days ago) • 2023-02-08 15:46:04 -0800
Engine • revision 248290d6d5
Tools • Dart 2.19.2 • DevTools 2.20.1

**Comment by Schmowser** [comment]
> @starshipcoder @RicardoRB @Hamitix can you share you flutter --version output ? I have an assumption, that the problem not in cluster version but in flutter version.

For me, it could be both. Downgraded flutter to the last time it worked with `flutter downgrade 3.3.8` and checked out a version that worked for me:

$ flutter --version
Flutter 3.3.8

with Dependencies
```
flutter_map 3.0.0
flutter_map_marker_cluster 1.0.1
```

Had to do a `flutter clean` (also make sure that pubspec.lock holds the versions you want) and rerun, then the Markers could be tapped again. Neither updating cluster version to 1.0.2 on Flutter 3.3.8 nor updating Flutter to 3.7.4 with any 1.0.x cluster version works for me. Hope that helps!

**Comment by Hamitix** [comment]
My flutter --version : 
Flutter 3.3.10 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 135454af32 (10 weeks ago) • 2022-12-15 07:36:55 -0800
Engine • revision 3316dd8728
Tools • Dart 2.18.6 • DevTools 2.15.0

-- 
With dependencies : 
flutter_map: ^3.1.0
flutter_map_marker_cluster: 1.0.2

-- 
Worked fine when downgrading to "flutter_map_marker_cluster: 1.0.1"

**Comment by dsanchez-neotreks** [comment]
Was anyone able to get this working again? For some reason it is not working on Android but on iOS works just fine. 
Does not work with current configuration on Android

Flutter 3.7.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 7048ed95a5 (4 weeks ago) • 2023-02-01 09:07:31 -0800
Engine • revision 800594f1f4
Tools • Dart 2.19.1 • DevTools 2.20.1


flutter_map: ^3.1.0
flutter_map_marker_cluster: 1.0.1

**Comment by keehoo** [comment]
For me it doesn't work on iOS as well. 1.0.1. Will check 1.0.2 and get back if it works there.

**Comment by enricostrijks** [comment]
Downgrade the package to 1.0.1 and make sure the pubspec.lock also holds 1.0.1, this was the fix for me!

**Comment by codscino** [comment]
> Downgrade the package to 1.0.1 and make sure the pubspec.lock also holds 1.0.1, this was the fix for me!

Done the same + downgrading to flutter 3.3.10 and now it works

**Comment by SeCo89** [comment]
Hi there, downgrading doesn't work for me. Is there an update to this problem using newer flutter versions? 
Thanks in advance

**Comment by fabiosebastiano-vertical** [comment]
Hi, have the same problem here and I still trying to figure out the solution!

**Comment by blablabla123-cell** [comment]
Same issue, had to remove clusters

**Comment by MohsenHaydari** [comment]
You should take a look at the code below
```
  marker: marker,
        onTap: _onMarkerTap(marker),
        onHover: (bool value) => _onMarkerHover(marker, value),
        buildOnHover: widget.options.popupOptions?.buildPopupOnHover ?? false,
        hoverOnTap: () => widget.options.onMarkerTap!(marker),
```
```
class MarkerWidget extends StatelessWidget {
  final MarkerNode marker;
  final VoidCallback onTap;
  final Function(bool)? onHover;
  final bool buildOnHover;
  final Function()? hoverOnTap;

  MarkerWidget({
    required this.marker,
    required this.onTap,
    this.onHover,
    this.buildOnHover = false,
    this.hoverOnTap,
  }) : super(key: marker.key ?? ObjectKey(marker.marker));

  @override
  Widget build(BuildContext context) {
    final m = marker.builder(context);
    return buildOnHover
        ? GestureDetector(
            behavior: HitTestBehavior.opaque,
            onTap: hoverOnTap ?? onTap,
            child: onHover != null
                ? MouseRegion(
                    onEnter: (_) => onHover!(true),
                    onExit: (_) => onHover!(false),
                    child: m,
                  )
                : m)
        : m;
  }
}
```

**Comment by tfreebern2** [comment]
We are having the same issue at work. 

Works fine with `flutter_map 3.1.0` and `flutter_map_marker_cluster 1.0.1` but when we upgrade to `flutter_map_marker_cluster 1.0.2` the `onMarkerTap` function no longer gets called.

[✓] Flutter (Channel stable, 3.3.7, on macOS 13.2.1 22D68 darwin-x64, locale en-US)

```
  flutter_map:
    dependency: "direct main"
    description:
      name: flutter_map
      url: "https://pub.dartlang.org"
    source: hosted
    version: "3.1.0"
  flutter_map_marker_cluster:
    dependency: "direct main"
    description:
      name: flutter_map_marker_cluster
      url: "https://pub.dartlang.org"
    source: hosted
    version: "1.0.2"
 ```

**Comment by MohsenHaydari** [comment]
> buildOnHover: widget.options.popupOptions?.buildPopupOnHover ?? false,

when you lock at the code 
```        
 buildOnHover: widget.options.popupOptions?.buildPopupOnHover ?? false,

```
onTap only work when 
```
widget.options.popupOptions?.buildPopupOnHover == True
````
```
 return buildOnHover
        ? GestureDetector(
            behavior: HitTestBehavior.opaque,
            onTap: hoverOnTap ?? onTap,
            child: onHover != null
                ? MouseRegion(
                    onEnter: (_) => onHover!(true),
                    onExit: (_) => onHover!(false),
                    child: m,
                  )
                : m)
        : m;
```

**Comment by proformance** [comment]
I had the same issue.

@MohsenHaydari is right on which part is the problematic code, I'm not sure what could break/be optimized by removing that if-statement, but it is clearly broken in pub.dev

Here is a solution in the meantime:

```
  flutter_map_marker_cluster:
    git:
      url: https://github.com/proformance/flutter_map_marker_cluster/
      ref: 999cd68
```

I'll submit a PR so the code change can be discussed

**Comment by jander96** [comment]
Same problem. Just dowgrade to 1.0.1 in pubspect,yaml and change version in pubspect.lock too

**Comment by LazuLitee** [comment]
This is still an open issue.

**Comment by LazuLitee** [comment]
This is still an open issue.

**Comment by enricostrijks** [comment]
It seems to be fixed in 1.1.1, the only problem left is that buildOnHover does not work. In https://github.com/lpongetti/flutter_map_marker_cluster/pull/175 it should be fixed if the maintainer merges this pr.
